src/snabbkaffe.erl

```diff
@@ -663,19 +666,21 @@ dump_trace(Trace) ->
   Filename = integer_to_list(os:system_time()) ++ ".log",
   FullPath = filename:join([CWD, "snabbkaffe", Filename]),
   Pretty = os:getenv("SNK_PRETTY_PRINT_DUMP") =/= false,
+  Formatter = fun(Evt) -> pp_trace(Pretty, Evt) end,
   filelib:ensure_dir(FullPath),
   {ok, Handle} = file:open(FullPath, [write]),
   try
-    lists:foreach(fun(I) ->
-                      dump_trace_event(Pretty, Handle, I)
-                  end, Trace)
+    dump_trace(Handle, Trace, Formatter)
   after
     file:close(Handle)
   end,
   FullPath.
+-else.
+dump_trace(Trace) ->
+  dump_trace(standard_io, Trace).
```

Comment:
I feel like this part is getting overly complicated. 
There is no real need to pretty-print the snabbkaffe trace when running under Concuerror: concuerror keeps its own trace, which is more detailed anyway. 
So it should be sufficient to simply print the events as is, via `lists:foreach(fun(Event) -> io:format("~p", [Event]) end, Trace)` here.
If the code is refactored like this, then there won't be any need to pass formatter and IO device to `dump_trace` function.

Reply:
I  handled it the same way (using `lists:foreach/2` and `io:format/3`) at https://github.com/kafka4beam/snabbkaffe/pull/58/files#diff-617f7c569331707a2095637219f56ce9a8428d6829f6c1fcf83255f0af95af8dR715

But I agree, it can be done directly in `dump_trace/1` at line 680.

Reply:
fixed